pkg/prometheus/server/operator.go

```diff
@@ -1318,6 +1318,10 @@ func addAlertmanagerEndpointsToStore(ctx context.Context, store *assets.StoreBui
 		if err := store.AddSigV4(ctx, namespace, am.Sigv4); err != nil {
 			return fmt.Errorf("alertmanager %d: %w", i, err)
 		}
+
+		if err := store.AddTLSConfig(ctx, namespace, am.TLSConfig); err != nil {
```

Comment:
Curious... didn't we need to add the TLS config to cache even before this PR, Did we miss that completely?

Reply:
TLSConfig  has been added to the cache.